framework/src/st_things/things_stack/framework/things_req_handler.c

```diff
@@ -190,6 +190,105 @@ OCEntityHandlerResult send_response(OCRequestHandle request_handle, OCResourceHa
 	return eh_result;
 }
 
+/**
+ * This functions will replace the get_ap_scan_result(~~)
+ */
+static OCEntityHandlerResult convert_ap_infor(things_resource_s *target_resource, access_point_info_s **p_list, int list_cnt)
+{
+
+	OCEntityHandlerResult eh_result = OC_EH_ERROR;
+
+	things_representation_s *rep = NULL;
+
+	if (!(list_cnt == 0 || p_list == NULL)) {
```

Comment:
Positive state is more easy to understand.
```
((list_cnt != 0) && (p_list != NULL))
or
if (list_cnt ==0) {
    ERROR_LOG("blahblah\n");
    return eh_result;
}

if (p_list == NULL) {
     ERROR_LOG("blahblah\n");
     return eh_result;
}
```
But this function doesn't have a else state, second is good option to reduce depth of branch